xcm/src/v3/traits.rs

```diff
@@ -539,3 +539,22 @@ pub fn send_xcm<T: SendXcm>(
 	let hash = T::deliver(ticket)?;
 	Ok((hash, price))
 }
+
+pub enum QueryResponseStatus<BlockNumber> {
+	Finished { response: Response, at: BlockNumber },
+	Pending,
+	NotFound,
+}
+
+/// Represents the
+pub trait XcmQueryHandler {
```

Comment:
I don't think the trait and the enum above should be located in this file -- even the `OnResponse` trait isn't. It should be moved to a new file under `xcm-executor/src/traits`.